### PR TITLE
Consent Engine: single configurable UID claim, drop owner_email (#477)

### LINKED ISSUE #477: Consent Engine: replace userEmail/OwnerEmail with a single configurable UID claim
### Current Limitation

The consent-engine (`exchange/consent-engine/`) identifies a data owner with **two identifiers that are always the same string** — `OwnerID` and `OwnerEmail` — and hardwires `email` as the token claim used for authorization. This does not fit a Digital Public Infrastructure model, where a user should be identified by a **single canonical UID** across the whole system.

Specifically:

- **`OwnerID == OwnerEmail` always.** The orchestration-engine sets `ownerEmail := dataOwnerID` (`federator.go:415`) and writes the same string into both `OwnerID` and `OwnerEmail` on the consent request (`federator.go:438-439`).
- **The consent record persists both.** `ConsentRecord` stores `owner_email`, and the active-consent unique index spans `(owner_id, owner_email, app_id)` (`v1/models/consent.go`). The create DTO requires both (`consent_service.go:387-392`).
- **The matching claim is hardcoded to `email`.** `VerifyTokenAndExtractEmail` reads `claims["email"]` (`v1/auth/jwt_verifier.go:283`); the middleware stores it as `userEmail` (`v1/middleware/auth.go`), and the portal handler authorizes by comparing `consent.OwnerEmail == userEmail` (`v1/handlers/portal_handler.go:90,163`) — i.e. against the email, not the UID, and with no way to configure which claim carries the identity.

As noted in `docs/FLOW-graphql-policy-consent-and-esignet-gap.md` (§5–§6), the `ownerEmail` label is a lie about the data it holds (it is frequently a NIC), and this identifier-namespace conflation must be resolved before eSignet citizen authentication can be integrated.

### Suggested Improvement

Remove the `userEmail`/`OwnerEmail` concept from the consent-engine and rely on a **single UID (`OwnerID`)**, with configuration declaring **which token claim maps to that UID**.

**Config**
- Add a subject-claim setting (e.g. `IDP_SUBJECT_CLAIM`, default `sub`) to `IDPConfig` (`internal/config/config.go`); thread it into `JWTVerifierConfig` in `main.go`. Document it in `README.md` / `.env.example`.

**Token verification & authorization**
- Replace `VerifyTokenAndExtractEmail` with a `VerifyTokenAndExtractSubject` that reads `claims[SubjectClaim]` instead of hardcoded `claims["email"]`.
- Rename the middleware context identity (`userEmailKey` → subject/ownerID; `GetUserEmailFromContext` → `GetUserSubjectFromContext`).
- Portal handlers authorize with `consent.OwnerID == subject` (drop the `OwnerEmail` comparison); `UpdatedBy` becomes the subject/UID.

**Model & persistence**
- Remove `OwnerEmail` from `ConsentRecord`, `ConsentRequirement`, `ConsentResponsePortalView`, and the create DTO.
- Rebuild the active-consent unique index on `(owner_id, app_id)`; drop `idx_consent_records_owner_email`.
- Provide a manual SQL migration to drop the `owner_email` column and the old indexes (`AutoMigrate` will not drop them).

**Service & internal API**
- Drop the `ownerEmail` parameter/branch from `GetConsentInternalView`; lookups key on `(ownerID, appID)` only.
- Remove the `ownerEmail` query param and required-field validation from the internal `GetConsent` handler and `validateCreateConsentRequest`.
- Update affected tests.

**Cross-service ripple (follow-up)**
- Orchestration-engine should stop sending `OwnerEmail` and seed `OwnerID` from the same claim the CE is configured to match. CE and OE must agree on which claim = the canonical UID (relevant to the eSignet PSUT/`sub` alignment described in the design docs).

### Version

commit 320305e1

### Additional Context

Related design docs in the repo:
- `docs/DESIGN-esignet-citizen-authentication.md`
- `docs/FLOW-graphql-policy-consent-and-esignet-gap.md` (see §5–§7 on the identifier-namespace conflation and the canonical-UID precondition for eSignet)

This change is scoped to `exchange/consent-engine/` and is a precondition for gating consent approval on a verified citizen identity.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

In a Digital Public Infrastructure setup a user should be identified system-wide by a single canonical UID, not by an email. Today the consent-engine stores an `OwnerEmail` alongside `OwnerID` (the two are always the same string), and it hardwires the JWT `email` claim as the identity used to authorize portal actions.

This PR removes the email-based identity from the consent-engine and makes the token claim that carries the owner UID configurable. It is split into two commits: (1) remove email, (2) add the configurable claim.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

> Breaking: the `owner_email` column and the `ownerEmail` API surface are removed; existing databases require a manual migration (see Deployment Notes).

## Changes Made

**Commit 1 — `refactor: identify data owner by single UID, drop owner_email`**
- Removed `OwnerEmail` from `ConsentRecord`, `ConsentRequirement`, and `ConsentResponsePortalView`. The active-consent partial unique constraint is now keyed on `(owner_id, app_id)`.
- Removed the `ownerEmail` lookup path from the internal `GET /internal/api/v1/consents` handler and from `GetConsentInternalView` (lookups key on `(ownerId, appId)` only).
- Portal authorization now compares `consent.OwnerID` against the token subject instead of comparing `owner_email` to the `email` claim.
- Verifier extracts the owner UID from the standard OIDC `sub` claim (`VerifyTokenAndExtractSubject`); middleware context renamed to `ownerSubject` (`GetOwnerSubjectFromContext`).
- Added `v1/migrations/drop_owner_email.sql` for existing databases and updated the base migration for fresh installs.

**Commit 2 — `feat: make the owner-UID token claim configurable`**
- Added `IDP_SUBJECT_CLAIM` (default `sub`) → `IDPConfig.SubjectClaim` → `JWTVerifierConfig.SubjectClaim`, threaded through `main.go` and logged at startup.
- `VerifyTokenAndExtractSubject` reads the configured claim, falling back to `DefaultSubjectClaim` (`sub`) when unset — so each IdP can declare which claim carries the UID (e.g. `sub`, `email`, `individual_id`).
- Documented the variable in `README.md` and `.env.example`.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

Unit tests updated/added: subject extraction defaults to `sub`, uses a configured claim, and fails when the configured claim is missing; middleware/handler tests updated to the subject/UID model. Full `go build`, `go vet`, and `go test ./...` pass; both commits passed the repo pre-commit validation (quality + build + tests).

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #477

## Additional Notes

Scope is limited to `exchange/consent-engine/`. The orchestration-engine still sends an `ownerEmail` field and seeds `OwnerID` positionally from `extractedArgs[0]`; the consent-engine now ignores the extra field. Aligning OE to seed `OwnerID` from the same claim the CE matches is a tracked follow-up (see `docs/FLOW-graphql-policy-consent-and-esignet-gap.md` §7), and is the precondition for the eSignet citizen-authentication work.

## Deployment Notes

- **Manual DB migration required for existing databases.** `AutoMigrate` does not drop columns/indexes, so run `v1/migrations/drop_owner_email.sql` to drop the `owner_email` column, drop `idx_consent_records_owner_email`, and rebuild `idx_consent_active_unique` on `(owner_id, app_id, status)`. The migration aborts if duplicate active consents exist that previously differed only by `owner_email`; resolve those first.
- **Config:** set `IDP_SUBJECT_CLAIM` to the claim that carries the owner UID for your IdP (defaults to `sub`). Ensure the value in that claim matches the `owner_id` seeded when consents are created.

### CODE REVIEW

File: exchange/consent-engine/v1/migrations/add_consent_unique_tuple_constraint.sql [hunk 1]
```diff
@@ -27,14 +27,13 @@ CREATE TABLE IF NOT EXISTS consent_records (
 );
 
 -- Create partial unique index for active consents (pending or approved)
--- This enforces that only one active consent can exist per (owner_id, owner_email, app_id)
-CREATE UNIQUE INDEX IF NOT EXISTS idx_consent_active_unique 
-    ON consent_records(owner_id, owner_email, app_id, status)
+-- This enforces that only one active consent can exist per (owner_id, app_id)
+CREATE UNIQUE INDEX IF NOT EXISTS idx_consent_active_unique
+    ON consent_records(owner_id, app_id, status)
     WHERE status IN ('pending', 'approved');
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/consent-engine/v1/migrations/add_consent_unique_tuple_constraint.sql:34]: Resolved in 50c0602cc9f8df5642802ccb1781fae62b7e0aa8. The base migration now creates the index on `(owner_id, app_id)` with `status` kept only in the `WHERE` filter, so at most one active (pending OR approved) consent can exist per `(owner_id, app_id)`. Good catch — this was a latent bug present before this PR too.

File: exchange/consent-engine/v1/migrations/drop_owner_email.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+--              that previously differed only by owner_email, the unique index below
+--              will fail to create. Resolve those duplicates before running.
+
+BEGIN;
+
+-- Rebuild the partial unique index without owner_email
+DROP INDEX IF EXISTS idx_consent_active_unique;
+CREATE UNIQUE INDEX IF NOT EXISTS idx_consent_active_unique
+    ON consent_records(owner_id, app_id, status)
+    WHERE status IN ('pending', 'approved');
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/consent-engine/v1/migrations/drop_owner_email.sql:27]: Resolved in 50c0602cc9f8df5642802ccb1781fae62b7e0aa8. The rebuilt index now keys on `(owner_id, app_id)` only. I also updated the migration's duplicate-warning note: the stricter key can now fail if an existing DB has a coexisting pending and approved row for the same `(owner_id, app_id)`, so those must be resolved before running.

File: exchange/consent-engine/v1/models/consent.go [hunk 3]
```diff
@@ -8,18 +8,17 @@ import (
...
 // - The most recently created record should have status 'pending' or 'approved' (if active)
 type ConsentRecord struct {
 	// ConsentID is the unique identifier for the consent record
 	ConsentID uuid.UUID `gorm:"column:consent_id;type:uuid;primaryKey;default:gen_random_uuid()" json:"consent_id"`
-	// OwnerID is the unique identifier for the data owner
+	// OwnerID is the single canonical identifier (UID) for the data owner.
+	// The whole system relies on this UID to identify a user; a token claim
+	// (configurable) is matched against it during portal authorization.
 	// Part of conditional unique constraint for active consents (pending/approved)
 	OwnerID string `gorm:"column:owner_id;type:varchar(255);not null;index:idx_consent_records_owner_id;index:idx_consent_records_owner_app;uniqueIndex:idx_consent_active_unique,where:status = 'pending' OR status = 'approved'" json:"owner_id"`
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply exchange/consent-engine/v1/models/consent.go:21]: Resolved in 50c0602cc9f8df5642802ccb1781fae62b7e0aa8. Removed the `uniqueIndex:idx_consent_active_unique` option from the `Status` field's gorm tag, so `AutoMigrate` now builds the index on `(owner_id, app_id)` only. The `where:status = 'pending' OR status = 'approved'` clause remains on the `owner_id`/`app_id` tags, preserving the partial-index filter. (The integration init SQL had the same issue and was corrected too.)

**Review by ginaxu1 — APPROVED** [review]
